src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java

```diff
@@ -0,0 +1,210 @@
+        final Properties props = new Properties();
+        try {
+            final Session session = Session.getDefaultInstance(props, null);
+            store = session.getStore("imaps");
+            store.connect(host, username, password);
+        } catch (final MessagingException e) {
+            throw new StopTestException("Cannot connect to mailserver");
+        }
+    }
+
+    /**
+     * | ensure | mail received with subject | <i>subject</i> | from | <i>receiver</i> |
+     *
+     * @param subject
+     * @return
+     */
+    public String mailReceivedWithSubjectFrom(final String subject, final String receiver) {
+        return getMailText(new SearchParameters(subject, receiver, HOURS_BACK));
+    }
+
+    /**
+     * | $result= | extract text with | <i>regex</i> | from mail with | <i>subject</i> | from | <i>receiver</i> |
+     */
+    public String extractTextWithFromMailWithFrom(final String regex, final String subject, final String receiver) {
+        final String text = getMailText(new SearchParameters(subject, receiver, HOURS_BACK));
+        return getText(text, regex);
+    }
+
+    private String getText(final String text, final String regexpattern) {
+        LOGGER.debug("Text found: " + text);
+        LOGGER.debug("Regex: " + regexpattern);
+        final Pattern pattern = Pattern.compile(regexpattern);
+        final Matcher matcher = pattern.matcher(text);
+        final boolean isMatch = matcher.find();
+        if (isMatch) {
+            return matcher.group(1);
+        }
+        throw new SlimFixtureException(false, format("No match found in text %s with regex %s", text, regexpattern));
+    }
+
+    private String getMailText(SearchParameters params) {
+        try {
+            final Folder inbox = getInboxFolder();
+            Message msg = getRecentMessagesMatching(inbox, params);
+            return getBody(msg);
+        } catch (final Exception ex) {
+            throw new SlimFixtureException(false, "No message found with search params: " + params.toString(), ex);
+        }
+    }
+
+    private Folder getInboxFolder() throws MessagingException {
+        final Folder inbox = store.getFolder("inbox");
+        inbox.open(Folder.READ_WRITE);
+        return inbox;
+    }
+
+    private Message getRecentMessagesMatching(Folder inbox, SearchParameters params) {
+        List<Message> mails = getMessagesMatchingAndWaitUntil(inbox, params);
+        Collections.reverse(mails);
+        return getMostRecentMessage(mails);
+    }
+
+    private Message getMostRecentMessage(List<Message> mails) {
+        try {
+            for (final Message mail : mails) {
+                if (mail.getSentDate().after(HOURS_BACK)) {
+                    return mail;
+                }
+            }
+        } catch (MessagingException ex) {
+            throw new RuntimeMessageException(ex);
+        }
+        throw new RuntimeIOException();
```

Comment:
Why not throw a SlimFixtureException stating that no messages were present sent after HOURS_BACK?
This provides more background about the limit used at runtime.